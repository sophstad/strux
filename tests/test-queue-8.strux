int main() {
	Queue::num q = new Queue::num();
	q.add(1.0);
	q.add(3.0);
	q.add(5.0);
	num i = q.peek();
	print(i);
	Queue::int q2 = new Queue::int();
	q2.add(3);
	q2.add(4);
	q2.add(5);
	int k = q2.peek();
	print(k);
	Queue::string q3 = new Queue::string();
	q3.add("mil");
	q3.add("soph");
	q3.add("kofi");
	q3.add("josh");
	string top = q3.peek();
	print(top);
	Queue::bool q4 = new Queue::bool();
	q4.add(true);
	q4.add(false);
	q4.add(false);
	q4.add(false);
	bool first = q4.peek();
	print(first);
	return 0;
}
