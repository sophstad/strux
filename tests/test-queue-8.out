1.000000
3
mil
1